include/networkit/algebraic/CSRGeneralMatrix.hpp

```diff
@@ -69,20 +50,22 @@ class CSRGeneralMatrix {
      * Constructs the CSRGeneralMatrix with size @a dimension x @a dimension.
      * @param dimension Defines how many rows and columns this matrix has.
      * @param zero The zero element (default = 0).
+     * @param isSorted If the matrix representation should uses sorted vectors.
      */
-    CSRGeneralMatrix(count dimension, ValueType zero = 0)
+    CSRGeneralMatrix(count dimension, ValueType zero = 0, bool isSorted = true)
```

Comment:
As discussed in #1146 and #1158, I'd suggest not complicating the constructor with additional parameters. Instead, as done in #1158, we could add some getter and setter functions for the `isSorted` variable.

Reply:
Would setting the sorted value not require unnecessary effort, if it is set from false to true? Or should the matrix be sorted by default and the setter only allows to create an unsorted version. In this case I would suggest a makeUnsorted() function or similar, instead of a setter-function.

Reply:
We could have the matrix sorted by default. If some user does not require that functionality, we could allow them to turn it off with some `makeUnosrted()` function as you suggest. What we should avoid is polluting the constructors with variables that control an internal behavior of the data structure.

Reply:
Latest commit includes the makeUnsorted function and the old constructors